### PR TITLE
Use new ScatterSwap implementation to only initialize the Hasher once

### PR DESCRIPTION
**Pull request by mullen3**

### CODE REVIEW

File: Gemfile
Comment (mullen3) [inline Gemfile]: Update me once we merge in the scatter_swap changes to master.

File: lib/obfuscate_id.rb
Comment (mullen3) [inline lib/obfuscate_id.rb:37]: I changed the naming convention here because the `obfuscate_id` method name was already taken and I wanted to add another class method for performing the obfuscations.

File: lib/obfuscate_id.rb
Comment (SempervirensSoftware) [inline lib/obfuscate_id.rb:25]: This is interesting. The find method in the ObfuscateID that my RubyMine is displaying has a totally different implementation for this `find` method. 

`deobfuscate_id` is neither used nor defined in my copy.  

It's showing as 0.0.3

File: lib/obfuscate_id.rb
Comment (SempervirensSoftware) [reply lib/obfuscate_id.rb:37]: There is also a an instance method that wraps this defined below. You'll need to update that one to us this new name, and potentially rename it as well.

It feels safest to maybe alias this with the old one, because who know who might be calling this static method somewhere

File: lib/obfuscate_id.rb
Comment (SempervirensSoftware) [inline lib/obfuscate_id.rb:12]: These `hide` and `show` methods need to be updated to the new method signatures on the ScatterSwap methods. Looks like only one param required.

File: lib/obfuscate_id.rb
Comment (SempervirensSoftware) [inline lib/obfuscate_id.rb:9]: This is going to be so much more efficient!

**Review by SempervirensSoftware — CHANGES_REQUESTED** [review]

File: lib/obfuscate_id.rb
Comment (mullen3) [reply lib/obfuscate_id.rb:12]: I think this is still right. I kept the old `ScatterSwap` static methods that take these params. See scatter_swap.rb for the definition of those methods.

File: lib/obfuscate_id.rb
Comment (mullen3) [reply lib/obfuscate_id.rb:25]: Interesting... I'll look into this.

File: lib/obfuscate_id.rb
Comment (mullen3) [reply lib/obfuscate_id.rb:37]: Sounds good on both fronts. My refactor in RubyMine must have missed this.

File: lib/obfuscate_id.rb
Comment (mullen3) [reply lib/obfuscate_id.rb:25]: Ok so I actually forked off of the latest release, `v0.2.0`. It appears that they fixed a bug with locking through `find` and added a `find` option that allows you to to skip the obfuscation. I'd be fine with taking the new changes while we're upgrading.

Comment (mullen3) [reply lib/obfuscate_id.rb:25]: The other changes are similar bugfixes in the `reload` method.

File: lib/obfuscate_id.rb
Comment (mullen3) [reply lib/obfuscate_id.rb:37]: Done.

File: lib/obfuscate_id.rb
Comment (SempervirensSoftware) [reply lib/obfuscate_id.rb:12]: got it

File: lib/obfuscate_id.rb
Comment (SempervirensSoftware) [reply lib/obfuscate_id.rb:25]: Sounds good. That `find` option would have come in handy in the past too

File: Gemfile
Comment (SempervirensSoftware) [reply Gemfile]: Last TODO

**Review by SempervirensSoftware — COMMENTED** [review]
Looks good. I'll approve when you switch over to the master branch of our scatter_swap fork

**Review by SempervirensSoftware — APPROVED** [review]

**Pull request merged**
